libraries/listen.rb

```diff
@@ -34,8 +34,8 @@ def self.converted_listen_ports_and_addresses(node)
       return [] unless node['apache']['listen_ports'] || node['apache']['listen_addresses']
       Chef::Log.warn "node['apache']['listen_ports'] and node['apache']['listen_addresses'] are deprecated in favor of node['apache']['listen']. Please adjust your cookbooks"
 
-      node['apache']['listen_addresses'].uniq.each_with_object([]) do |address, listen|
-        node['apache']['listen_ports'].uniq.each do |port|
+      (node['apache']['listen_addresses'] || []).uniq.each_with_object([]) do |address, listen|
+        (node['apache']['listen_ports'] || []).uniq.each do |port|
```

Comment:
I assumed the defaults we removed because they did not want to show the old way of doing this and wanted to retire it.  It makes sense to just have a blank array and do nothing I feel.   If you add in defaults people who do not use the new syntax or full old syntax could end up with new behavior from the update.  Also this code is now tested because of  the addition of the code segment thats below.  That sets the inner conditional so the code path is hit.  

```
default['apache']['listen_ports'] = ['80']
```